File: be/src/main/java/com/ymc/paper/domain/TranslationStatus.java
package com.ymc.paper.domain;

/** API의 translationStatus. 문서 언어와 컴파일 상태에서 계산하며 DB에 따로 저장하지 않는다. */
public enum TranslationStatus {
    NOT_APPLICABLE,
    PENDING,
    READY,
    FAILED;

    /**
     * 영어가 아니면 번역 대상이 아니다. 영어인데 컴파일 상태가 없으면 아직 요청 전이라 PENDING이다 —
     * 발행 실패 뒤 재전달을 기다리는 창에서 FAILED로 답하면 FE가 폴링을 멈춘다.
     */
    public static TranslationStatus of(String sourceLanguage, CompileStatus compileStatus) {
        if (!"en".equals(sourceLanguage)) {
            return NOT_APPLICABLE;
        }
        if (compileStatus == null) {
            return PENDING;
        }
        return switch (compileStatus) {
            case REQUESTED -> PENDING;
            case COMPLETED -> READY;
            case FAILED -> FAILED;
        };
    }
}

File: fe/src/routes/study/translationStatus.ts
// 학습 화면의 번역 상태 처리. 폴링은 PENDING일 때만, 툴팁은 아트보드 문구 그대로.
import type { TranslationStatus } from '../../api/types';

/** 컴파일은 분 단위라 서재의 2초 폴링보다 느슨하게 둔다. */
export const TRANSLATION_POLL_MS = 5000;

export function translationRefetchInterval(status: TranslationStatus | undefined): number | false {
  return status === 'PENDING' ? TRANSLATION_POLL_MS : false;
}

const REASON: Partial<Record<TranslationStatus, string>> = {
  NOT_APPLICABLE: '한국어 논문은 번역하지 않습니다',
  PENDING: '번역을 준비하고 있습니다',
  FAILED: '번역 생성에 실패했습니다',
};

/** 번역 버튼 비활성 사유. READY인데 번역 블록이 없는 경우(사이드카 없음·병합 0건)는 준비 안 됨 문구다. */
export function translationDisabledReason(status: TranslationStatus | undefined, hasTranslation: boolean): string {
  if (status === 'READY' && hasTranslation) return '';
  return (status && REASON[status]) ?? '이 논문은 번역이 준비되지 않았습니다';
}
